state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1530,6 +1546,92 @@ impl<T: Config> Pallet<T> {
 		)
 	}
 
+	pub fn pool_orderbook(
+		base_asset: any::Asset,
+		quote_asset: any::Asset,
+		orders: u32,
+	) -> Option<PoolOrderbook> {
```

Comment:
How come we are returning an option here instead of a result? If it were a result, we could return a `PoolDoesNotExist` error.
Same for `pool_orders`, `pool_info` etc..
But in `pool_liquidity_providers` we do return the `PoolDoesNotExist` error.  We should be consistent.